zb-db/src/test/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyCheckerTest.java

```diff
@@ -92,6 +160,70 @@ void shouldSucceedOnRealColumnFamily(@TempDir final File tempDir) throws Excepti
     db.close();
   }
 
+  @Test
+  void shouldFindByPrefix(@TempDir final File tempDir) throws Exception {
+    // given
+    final var db = DefaultZeebeDbFactory.<TestColumnFamilies>getDefaultFactory().createDb(tempDir);
+    final var txContext = db.createContext();
+
+    final var cf1Key = new DbCompositeKey<>(new DbLong(), new DbString());
+    cf1Key.first().wrapLong(1);
+    cf1Key.second().wrapString("suffix");
+    final var cf1 =
+        db.createColumnFamily(
+            TestColumnFamilies.TEST_COLUMN_FAMILY, txContext, cf1Key, DbNil.INSTANCE);
+
+    final var check =
+        new ForeignKeyChecker(
+            (ZeebeTransactionDb<?>) db, new ConsistencyChecksSettings(true, true));
+
+    // when -- key 1 exists in first column family
+    cf1.insert(cf1Key, DbNil.INSTANCE);
+
+    // then -- referring to key 1 by prefix does not throw
+    final var fk =
+        new DbForeignKey<>(
+            new DbLong(), TestColumnFamilies.TEST_COLUMN_FAMILY, MatchType.Prefix, (any) -> false);
```

Comment:
```suggestion
(any) -> false);
```
:clown_face: So much better!

Reply:
:thinking: I don't understand the suggested change. Are you suggesting adding a constructor to `DbForeignKey` that only accepts the `skip` predicate?

Reply:
No this, is related to https://github.com/camunda/zeebe/pull/8960#discussion_r832602990